docs/source/emulating-compiler-behavior.rst

```diff
@@ -0,0 +1,125 @@
+Emulating Compiler Behavior
+===========================
+
+When CBI processes a file, it tries to obey all of the arguments that it can
+see in the compilation database. Unfortunately, compilers often have behaviors
+that are not reflected on the command line (such as their default include
+paths, or compiler version macros).
+
+If we believe (or already know!) that these behaviors will impact the
+divergence calculation for a code base, we can use a configuration file to
+instruct CBI to append additional options when emulating certain compilers.
+
+.. attention::
+
+    If you encounter a situation that is not supported by CBI and which cannot
+    be described by our existing configuration files, please `open an issue`_.
+
+.. _`open an issue`: https://github.com/intel/code-base-investigator/issues/new/choose
+
+
+Motivating Example
+------------------
+
+The ``foo.cpp`` files in our sample code base include specialization that we
+have ignored so far, which selects a line based on the value of the
+:code:`__GNUC__` preprocessor macro:
+
+.. code-block:: cpp
+    :linenos:
+    :emphasize-lines: 6
+
+    // Copyright (c) 2024 Intel Corporation
+    // SPDX-License-Identifier: 0BSD
+    #include <cstdio>
+
+    void foo() {
+    #if __GNUC__ >= 13
+        printf("Using a feature that is only available in GCC 13 and later.\n");
+    #else
+        printf("Running the rest of foo() on the CPU.\n");
+    #endif
+    }
+
+This macro is defined automatically by all GNU compilers and is set based on
+the compiler's major version. For example, ``gcc`` version 13.0.0 would set
+:code:`__GNUC__` to 13. Checking the values of macros like this one can be
+useful when specializing code paths to workaround bugs in specific compilers,
+or when specializing code paths to make use of functionality that is only
+available in newer compiler versions.
+
+Let's take another look at the compilation database entry for this file:
+
+.. code-block:: json
+    :emphasize-lines: 14
```

Comment:
I'm not sure why, but Github is highlighting this line in red: can you double check this renders correctly?

Reply:
It renders correctly.  I think it might be that GitHub doesn't support these decorations.  If I render the file (in GitHub), this code block and the cpp code block above it aren't rendered at all.